### PR TITLE
Fix problem with using Ember Array wrappers as content in Select component

### PR DESCRIPTION
**Pull request by thangdinh**
### Problem
Currently we are using lodash to find the selection among the content in the select component:
`selection = _.find(content, valuePath, value)`
It does not work for array wrappers such as `ArrayProxy`, `ArrayController`, `DS.ManyArray`, for which we have to use `findProperty` (old version of `findBy` because of our current version of Ember here).

### Solution
Check if the content is a wrapper or not, if it is use `findProperty`, otherwise use `_.find`

@cyril-sf @bjcohen @Addepar/fire

### CODE REVIEW

File: tests/unit/select_test.coffee [hunk 1]
```diff
@@ -258,3 +258,70 @@ test "Show no-result message if has content but filtered content is empty", ->
       'Empty content block not displayed'
     ok isPresent(noResultSelector, selectElement),
       '"No result" message displayed'
+
+test 'optionValuePath with POJOs', ->
+  expect 1
+
+  obj1 = {name: 'reddit', value: 1}
+  obj2 = {name: 'red', value: 2}
+  data = [obj1, obj2]
```
Comment (bjcohen) [inline tests/unit/select_test.coffee]: Prototype extensions are off in these tests?

**Comment by bjcohen** [comment]
Is the case
```
obj1 = {name: 'reddit', value: {subvalue: '1'}}
obj2 = {name: 'red', value: {subvalue: '2'}}
data = [obj1, obj2]
select = @subject
  content: data
  optionLabelPath: 'name'
  optionValuePath: 'value.subvalue'
```

expected to work? The way it's written right now I don't think it does.

File: tests/unit/select_test.coffee [hunk 2]
```diff
@@ -258,3 +258,70 @@ test "Show no-result message if has content but filtered content is empty", ->
...
+  Ember.run ->
+    select.set 'value', 2
+
+  equal obj2, select.get('selection'),
+    'The right selection is retrieved'
+
+test 'optionValuePath with Ember Objects', ->
+  expect 1
+
+  Klass = Ember.Object.extend
```
Comment (jordansmith42) [inline tests/unit/select_test.coffee]: I like Klass, but might just want to call it ObjectStub or something

Comment (cyril-sf) [inline tests/unit/select_test.coffee]: I wrote that, used to do that in ruby.

File: tests/unit/select_test.coffee (hunk 1 above)
Comment (cyril-sf) [inline tests/unit/select_test.coffee]: no. This test was initially written to reproduce the problem, before I understood what the problem is.

But that's a good point and we should test that.

**Comment by thangdinh** [comment]
@bjcohen I think it works in the current code

**Comment by thangdinh** [comment]
I just added the test for that

**Comment by bjcohen** [comment]
It works with prototype extensions off?

File: tests/unit/select_test.coffee [hunk 3]
```diff
@@ -1,6 +1,7 @@
 select = null
 emptyContentSelector = '.ember-select-empty-content'
 noResultSelector = '.ember-select-no-results'
+hasExtendedPrototype = null
```
Comment (cyril-sf) [inline tests/unit/select_test.coffee]: unused

**Comment by thangdinh** [comment]
Is there any other concern? Should I change `Klass` @cyril-sf @jordansmith42 ? :)

**Comment by cyril-sf** [comment]
@bjcohen we ran the test manually with the prototype extension off.

We'll make sure that both cases are tested once we have converted the library to an Ember-CLI addon.

**Comment by cyril-sf** [comment]
LGTM

@thangdinh did you test it in Iverson?

**Pull request merged**